### PR TITLE
Manage Spending Accounts + Profile UI Improvements.

### PR DESCRIPTION
**Pull request by oliverheywood451**
# Description

- Added a section to payment methods that displays spending accounts.
- Refactored credit-card-display an address-display to allow transcluded action buttons ( a more flexible way to have read-only options for display components ).
- Generally improved the profile UI. 

Fixes https://github.com/ordercloud-api/ngx-shopper/issues/37

## Type of change

- [ x] New feature (non-breaking change which adds functionality)
- [ x] Refactor (change/optimize code without changing external behavior)

## Checklist:

- [ x] I have performed a self-review of my own code
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes
- [ x] I have updated the acceptance criteria on the task so that it can be tested

### CODE REVIEW

File: src/UI/Buyer/src/app/profile/containers/payment-list/payment-list.component.ts [hunk 1]
```diff
@@ -45,7 +58,7 @@ export class PaymentListComponent implements OnInit, OnDestroy {
     this.authorizeNetSerivce.CreateCreditCard(card).subscribe(
```
Comment (crhistianramirez) [inline src/UI/Buyer/src/app/profile/containers/payment-list/payment-list.component.ts]: I think its a bit too opinionated to have a specific credit card integration in the base app, I had meant to remove authNet completely. We should leave it up to the developer to decide which integration to use.

Comment (oliverheywood451) [reply src/UI/Buyer/src/app/profile/containers/payment-list/payment-list.component.ts]: I do think its important to showcase how to use credit cards on OC. It was a harder topic for me to learn. How can we create credit card functionality without some integration?

Comment (crhistianramirez) [reply src/UI/Buyer/src/app/profile/containers/payment-list/payment-list.component.ts]: To be clear I'm not saying we shouldn't allow usage of credit cards at all, we'll just exclude the non-ordercloud parts of using a credit card (authorization etc). I think this is one of the components that will benefit from being well commented. 

You can still show how list credit cards to select, create a payment with Type CreditCard and the CreditCardID, the only thing that would be mocked is that payment.Accepted = true will have to be done in the application and requires an elevated role. We should mention that this is something that is typically done by a payment processor once the credit card has been authorized.

Comment (crhistianramirez) [reply src/UI/Buyer/src/app/profile/containers/payment-list/payment-list.component.ts]: you can check out angular-buyer (angularjs) to see kind of how it was built there

**Pull request merged**